modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Monad.kt

```diff
@@ -48,15 +48,36 @@ interface Monad<F> : Selective<F> {
   fun <A, B> Kind<F, A>.followedByEval(fb: Eval<Kind<F, B>>): Kind<F, B> =
     flatMap { fb.value() }
 
+  @Deprecated(
+    "effectM is being renamed to flatTap",
+    ReplaceWith("flatTap")
+  )
   fun <A, B> Kind<F, A>.effectM(f: (A) -> Kind<F, B>): Kind<F, A> =
+    flatTap(f)
+
+  fun <A, B> Kind<F, A>.flatTap(f: (A) -> Kind<F, B>): Kind<F, A> =
     flatMap { a -> f(a).map { a } }
 
-  fun <A, B> Kind<F, A>.forEffect(fb: Kind<F, B>): Kind<F, A> =
+  fun <A, B> Kind<F, A>.productL(fb: Kind<F, B>): Kind<F, A> =
     flatMap { a -> fb.map { a } }
 
-  fun <A, B> Kind<F, A>.forEffectEval(fb: Eval<Kind<F, B>>): Kind<F, A> =
+  @Deprecated(
+    "forEffect is being renamed to productL",
+    ReplaceWith("productL")
```

Comment:
is this the proper syntax for replacements?

Reply:
It's consistent with `@Deprecated`/`ReplaceWith` as used by elsewhere in Arrow.

Reply:
Which is wrong lol

Reply:
As near as I can tell from the Kotlin docs the right syntax is `ReplaceWith("productL(fb)")`, does that look right?

Reply:
yeah!